build fails with aeson-2.0

```
ghcjs-base-stub                > Building library for ghcjs-base-stub-0.3.0.2..
ghcjs-base-stub                > [ 1 of 29] Compiling Data.JSString.Internal.Fusion.CaseMapping
ghcjs-base-stub                > [ 2 of 29] Compiling Data.JSString.Internal.Fusion.Common
ghcjs-base-stub                > [ 3 of 29] Compiling Data.JSString.Internal.Fusion.Types
ghcjs-base-stub                > [ 4 of 29] Compiling Data.JSString.Internal.Search
ghcjs-base-stub                > [ 5 of 29] Compiling GHCJS.Prim
ghcjs-base-stub                > [ 6 of 29] Compiling GHCJS.Internal.Types
ghcjs-base-stub                > [ 7 of 29] Compiling GHCJS.Concurrent
ghcjs-base-stub                > [ 8 of 29] Compiling Data.JSString.Internal.Type
ghcjs-base-stub                > [ 9 of 29] Compiling Data.JSString.Text
ghcjs-base-stub                > [10 of 29] Compiling Data.JSString.Internal.Fusion
ghcjs-base-stub                > [11 of 29] Compiling Data.JSString
ghcjs-base-stub                > [12 of 29] Compiling GHCJS.Types
ghcjs-base-stub                > [13 of 29] Compiling GHCJS.Foreign.Internal
ghcjs-base-stub                > [14 of 29] Compiling GHCJS.Foreign.Export
ghcjs-base-stub                > [15 of 29] Compiling GHCJS.Foreign.Callback.Internal
ghcjs-base-stub                > [16 of 29] Compiling GHCJS.Foreign.Callback
ghcjs-base-stub                > [17 of 29] Compiling GHCJS.Foreign
ghcjs-base-stub                > [18 of 29] Compiling JavaScript.Array.Internal
ghcjs-base-stub                > [19 of 29] Compiling JavaScript.Array
ghcjs-base-stub                > [20 of 29] Compiling JavaScript.Cast
ghcjs-base-stub                > [21 of 29] Compiling JavaScript.Object.Internal
ghcjs-base-stub                > [22 of 29] Compiling JavaScript.Object
ghcjs-base-stub                >             
ghcjs-base-stub                > /tmp/stack-71a3f8fe3d9f7ff2/ghcjs-base-stub-0.3.0.2/src/JavaScript/Object.hs:11:1: warning: [-Wunused-imports]
ghcjs-base-stub                >     The import of ‘JavaScript.Object.Internal’ is redundant
ghcjs-base-stub                >       except perhaps to import instances from ‘JavaScript.Object.Internal’
ghcjs-base-stub                >     To import instances alone, use: import JavaScript.Object.Internal()
ghcjs-base-stub                >    |        
ghcjs-base-stub                > 11 | import           JavaScript.Object.Internal (Object(..))
ghcjs-base-stub                >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ghcjs-base-stub                > [23 of 29] Compiling GHCJS.Marshal.Internal
ghcjs-base-stub                > [24 of 29] Compiling GHCJS.Marshal.Pure
ghcjs-base-stub                >             
ghcjs-base-stub                > /tmp/stack-71a3f8fe3d9f7ff2/ghcjs-base-stub-0.3.0.2/src/GHCJS/Marshal/Pure.hs:26:52: warning: [-Wunused-imports]
ghcjs-base-stub                >     The import of ‘Word’ from module ‘Data.Word’ is redundant
ghcjs-base-stub                >    |        
ghcjs-base-stub                > 26 | import           Data.Word (Word8, Word16, Word32, Word)
ghcjs-base-stub                >    |                                                    ^^^^
ghcjs-base-stub                > [25 of 29] Compiling GHCJS.Nullable
ghcjs-base-stub                > [26 of 29] Compiling GHCJS.Marshal
ghcjs-base-stub                >             
ghcjs-base-stub                > /tmp/stack-71a3f8fe3d9f7ff2/ghcjs-base-stub-0.3.0.2/src/GHCJS/Marshal.hs:157:40: error:
ghcjs-base-stub                >     • Couldn't match type: H.HashMap Text v0
ghcjs-base-stub                >                      with: Data.Aeson.KeyMap.KeyMap AE.Value
ghcjs-base-stub                >       Expected: AE.Object
ghcjs-base-stub                >         Actual: H.HashMap Text v0
ghcjs-base-stub                >     • In the first argument of ‘AE.Object’, namely
ghcjs-base-stub                >         ‘(H.fromList propVals)’
ghcjs-base-stub                >       In the first argument of ‘return’, namely
ghcjs-base-stub                >         ‘(AE.Object (H.fromList propVals))’
ghcjs-base-stub                >       In a stmt of a 'do' block: return (AE.Object (H.fromList propVals))
ghcjs-base-stub                >     • Relevant bindings include
ghcjs-base-stub                >         propVals :: [(Text, v0)] (bound at src/GHCJS/Marshal.hs:154:21)
ghcjs-base-stub                >     |       
ghcjs-base-stub                > 157 |                     return (AE.Object (H.fromList propVals))
ghcjs-base-stub                >     |                                        ^^^^^^^^^^^^^^^^^^^
ghcjs-base-stub                >             
ghcjs-base-stub                > /tmp/stack-71a3f8fe3d9f7ff2/ghcjs-base-stub-0.3.0.2/src/GHCJS/Marshal.hs:278:102: error:
ghcjs-base-stub                >     • Couldn't match type: Data.Aeson.KeyMap.KeyMap AE.Value
ghcjs-base-stub                >                      with: H.HashMap Text AE.Value
ghcjs-base-stub                >       Expected: H.HashMap Text AE.Value
ghcjs-base-stub                >         Actual: AE.Object
ghcjs-base-stub                >     • In the first argument of ‘H.toList’, namely ‘o’
ghcjs-base-stub                >       In the second argument of ‘mapM_’, namely ‘(H.toList o)’
ghcjs-base-stub                >       In a stmt of a 'do' block:
ghcjs-base-stub                >         mapM_
ghcjs-base-stub                >           (\ (k, v)
ghcjs-base-stub                >              -> convertValue v
ghcjs-base-stub                >                   >>= \ v' -> OI.setProp (JSS.textToJSString k) v' obj)
ghcjs-base-stub                >           (H.toList o)
ghcjs-base-stub                >     |       
ghcjs-base-stub                > 278 |       mapM_ (\(k,v) -> convertValue v >>= \v' -> OI.setProp (JSS.textToJSString k) v' obj) (H.toList o)
ghcjs-base-stub                >     |                                                                                                      ^
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # Changelog
 
+* Unreleased
+  - Add aeson 2 support
+
 * 0.3.0.0
   - Removed upper bounds from dependencies so the base library does not limit an application's choice of libraries.
   - Re-exports ghjs-base modules, so this package can replace ghcjs-base.

--- a/src/GHCJS/Marshal.hs
+++ b/src/GHCJS/Marshal.hs
@@ -26,7 +26,12 @@ import           Control.Monad
 import           Control.Monad.Trans.Maybe (MaybeT(..), runMaybeT)
 
 import qualified Data.Aeson as AE
+#if MIN_VERSION_aeson(2,0,0)
+import qualified Data.Aeson.Key as AEK
+import qualified Data.Aeson.KeyMap as H
+#else
 import qualified Data.HashMap.Strict as H
+#endif
 import           Data.Int (Int8, Int16, Int32)
 import qualified Data.JSString.Text as JSS
 import           Data.Scientific (Scientific, scientific, fromFloatDigits)
@@ -151,9 +156,15 @@ instance FromJSVal AE.Value where
             JSONObject  -> do
                 props <- OI.listProps (OI.Object r)
                 runMaybeT $ do
+                    let toKey =
+#if MIN_VERSION_aeson(2,0,0)
+                          AEK.fromText
+#else
+                          id
+#endif
                     propVals <- forM props $ \p -> do
                         v <- MaybeT (fromJSVal =<< OI.getProp p (OI.Object r))
-                        return (JSS.textFromJSString p, v)
+                        return (toKey $ JSS.textFromJSString p, v)
                     return (AE.Object (H.fromList propVals))
     {-# INLINE fromJSVal #-}
 instance (FromJSVal a, FromJSVal b) => FromJSVal (a,b) where
@@ -275,5 +286,11 @@ toJSVal_aeson x = cv (AE.toJSON x)
     convertValue (AE.Bool b)   = return (toJSBool b)
     convertValue (AE.Object o) = do
       obj@(OI.Object obj') <- OI.create
-      mapM_ (\(k,v) -> convertValue v >>= \v' -> OI.setProp (JSS.textToJSString k) v' obj) (H.toList o)
+#if MIN_VERSION_aeson(2,0,0)
+      let fromKey =
+            AEK.toText
+#else
+            id
+#endif
+      mapM_ (\(k,v) -> convertValue v >>= \v' -> OI.setProp (JSS.textToJSString $ fromKey k) v' obj) (H.toList o)
       return obj'

